### PR TITLE
Fix "Empty string supplied as input" bug

### LINKED ISSUE #628: DOMDocument::loadHTML(): Empty string supplied as input
DOMDocument::loadHTML(): Empty string supplied as input
 Type: PHP Warning Line: 178
 File:  /wp-content/plugins/fb-instant-articles/vendor/facebook/facebook-instant-articles-sdk-php/src/Facebook/InstantArticles/Transformer/Transformer.php

Is this a bug or just a null value?

### PR DESCRIPTION
**Pull request by abdusfauzi**
This PR:

Fixes #
Fix `PHP Warning:  DOMDocument::loadHTML(): Empty string supplied as input`, by verifying the content and making sure if it is empty, do not `transformString`.

Automattic/facebook-instant-articles-wp#628

### CODE REVIEW

File: class-instant-articles-post.php [hunk 1]
```diff
@@ -690,7 +690,9 @@ public function to_instant_article() {
 			Video::setDefaultCommentEnabled( $settings_publishing[ 'comments_on_media' ] );
 		}
 
-		$transformer->transformString( $this->instant_article, $this->get_the_content(), get_option( 'blog_charset' ) );
+		if ( '' != $this->get_the_content() ) {
```
Comment (everton-rosario) [inline class-instant-articles-post.php]: Is this happening to you?
I'd change this, so we call only once the `get_the_content()` method here.
Can you do it like this:
```
$the_content = $this->get_the_content();
if ( '' != $the_content ) {
   $transformer->transformString...
}
```

Comment (abdusfauzi) [inline class-instant-articles-post.php]: Cool. No problem 👌🏻

File: class-instant-articles-post.php [hunk 2]
```diff
@@ -689,8 +689,11 @@ public function to_instant_article() {
 			Image::setDefaultCommentEnabled( $settings_publishing[ 'comments_on_media' ] );
 			Video::setDefaultCommentEnabled( $settings_publishing[ 'comments_on_media' ] );
 		}
-
-		$transformer->transformString( $this->instant_article, $this->get_the_content(), get_option( 'blog_charset' ) );
+		
+		$the_content = $this->get_the_content();
+		if ( '' != $the_content ) {
```
Comment (vkama) [inline class-instant-articles-post.php]: Can you please use the Type::isTextEmpty() method for this? Ty

Comment (abdusfauzi) [inline class-instant-articles-post.php]: @vkama i'm sorry, i wish i understand how to use the suggested `Type:isTextEmpty()` 😰 Would you mind show me sample?

Comment (vkama) [inline class-instant-articles-post.php]: `
use Facebook\InstantArticles\Validators\Type;

if (!Type::isTextEmpty($the_content) { ... }
`

File: class-instant-articles-post.php [hunk 3]
```diff
@@ -691,7 +693,7 @@ public function to_instant_article() {
 		}
 		
 		$the_content = $this->get_the_content();
-		if ( '' != $the_content ) {
+		if (!Type::isTextEmpty($the_content)) {
```
Comment (vkama) [inline class-instant-articles-post.php]: good job ty

**Pull request merged**